### PR TITLE
wishcard create + manage page enhancements

### LINKED ISSUE #639: STORY: "Create a Wish Card" page - UI redesign, review validation rules

### LINKED ISSUE #640: STORY: "Create a Wish Card" page - default shipping address toggle

### LINKED ISSUE #641: STORY: "Create a Wish Card" page - item image uploader

### LINKED ISSUE #645: STORY: "Manage Wish Cards" -  UI redesign

### LINKED ISSUE #702: TASK: field level validation + error input scroll and focus
for create wishcard page, add field level validation + error input scroll and focus

### PR DESCRIPTION
**Pull request by stacysealky**
resolves #639 
- converted the wishCardCreate page to react
- new design completed
- added mobile responsiveness
- added reusable popovers and constants for form helper texts 
- backend changes (new api route and controller)
- multiple file submission instead of single file

resolves #640 
- added a switch toggle for default shipping address
- validation PENDING (will create another task) 

resolves #641 
- item image uploader UI completed

resolves #645 
- redesigned wish card manage page with tab panels 

resolves #702 
- form field level validation with different error messages
- error input auto focus and scroll  

Before:
screencapture-donate-gifts-wishcards-manage-2023-10-04-03_54_32

After:
screencapture-localhost-3000-wishcards-manage-2023-10-04-04_00_06

Before: 
screencapture-donate-gifts-wishcards-create-2023-10-03-18_32_26

After:
screencapture-localhost-3000-wishcards-create-2023-10-04-21_04_37

Validation error displays:
screencapture-localhost-3000-wishcards-create-2023-10-04-21_08_01

### CODE REVIEW

File: js/react/utils/constants.jsx [hunk 1]
```diff
@@ -4,4 +4,57 @@ const SIGNUP = 'SIGNUP';
 
 const GOOGLE_CLIENT_LIBRARY_URL = 'https://accounts.google.com/gsi/client';
```
Comment (Enubia) [inline js/react/utils/constants.jsx:5]: this can be a simple `.js` file, no need for a `.jsx` extension

File: package.json [hunk 2]
```diff
@@ -36,6 +36,9 @@
 	},
 	"homepage": "https://github.com/stacysealky/donategifts#readme",
 	"dependencies": {
+		"@mantine/core": "^7.1.2",
+		"@mantine/dropzone": "^7.1.2",
+		"@mantine/hooks": "^7.1.2",
```
Comment (Enubia) [inline package.json:41]: this is already installed in dev dependencies

Comment (Enubia) [inline package.json:41]: please add them to the dev dependencies, we prebuild our react scripts and only ship the compiled stuff

File: public/svg/upload-icon.svg
Comment (Enubia) [inline public/svg/upload-icon.svg:1]: https://fontawesome.com/v5/search?q=upload&o=r&m=free

we can use font awesome, I'd rather use this instead, unless it's something specific that you created

File: package.json [hunk 3]
```diff
@@ -36,6 +36,9 @@
 	},
 	"homepage": "https://github.com/stacysealky/donategifts#readme",
 	"dependencies": {
+		"@mantine/core": "^7.1.2",
```
Comment (Enubia) [inline package.json:39]: this is already installed in dev dependencies

File: js/react/utils/constants.jsx (hunk 1 above)
Comment (stacysealky) [reply js/react/utils/constants.jsx:5]: it was an existing file, but ok I will change it

File: public/svg/upload-icon.svg
Comment (stacysealky) [reply public/svg/upload-icon.svg:1]: hmm how do you use font awesome for the ::before pseudo class? 
`&::before {
			position: absolute;
			pointer-events: none;
			top: 0.5em;
			left: 1.5em;
			height: 25px;
			width: 25px;
			content: "";
			background-image: url("/svg/upload-icon.svg");
		}
`

Comment (Enubia) [reply public/svg/upload-icon.svg:1]: ah nvm then, didn't notice this pseudo element

Comment (Enubia) [reply public/svg/upload-icon.svg:1]: > hmm how do you use font awesome for the ::before pseudo class?
> 
> ```
> &::before {
> 	position: absolute;
> 	pointer-events: none;
> 	top: 0.5em;
> 	left: 1.5em;
> 	height: 25px;
> 	width: 25px;
> 	content: "";
> 	background-image: url("/svg/upload-icon.svg");
> }
> ```

but do we actually need this? can't we just add the font awesome icon and the text in the button?

File: js/react/utils/constants.jsx (hunk 1 above)
Comment (Enubia) [reply js/react/utils/constants.jsx]: sorry, must have missed that in a previous review

File: public/svg/upload-icon.svg
Comment (Enubia) [reply public/svg/upload-icon.svg:1]: nvm my previous comments, I saw what you did there with the input field, all good 👍

File: example.config.env [hunk 4]
```diff
@@ -11,7 +11,7 @@ NODE_ENV="development"
 
 SESS_NAME="sid"
 SESS_SECRET="something-super-duper-secret"
-SESS_LIFE=3600000
+SESS_LIFE=36000000
```
Comment (Enubia) [inline example.config.env:14]: please revert this back to the original value, I don't think we want the session to last for 600 hours

File: js/react/pages/wishcard/WishCardCreate.jsx [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+		const toast = new window.DG.Toast();
+
+		try {
+			await axios
+				.post('/api/wishcards', data, {
+					headers: {
+						'content-type': 'multipart/form-data',
+					},
+				})
+				.then(toast.show('Submission was successful!'));
```
Comment (Enubia) [inline js/react/pages/wishcard/WishCardCreate.jsx:237]:
```suggestion
			await axios
				.post('/api/wishcards', data, {
					headers: {
						'content-type': 'multipart/form-data',
					},
				})
			toast.show('Submission was successful!');
```

**Review by Enubia — APPROVED** [review]

**Pull request merged**